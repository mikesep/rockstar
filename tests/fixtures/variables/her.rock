Her workout takes time and energy
give back energy with time

Her times are 5
Alice is 3

say "pronoun"
(her = Alice = 3)
say her times 7
say her times nothing
say her times "her"
say her times her
say her times him
say her times her times
say her times her workout taking 5, and 6
say ""

say "common variable"
say her times
say her times is 4
say her times is her
say her times is them
say her times are bigger than 8
say her times times her
say her times times her times
say ""

say "assignment"
Alice is a great leader (156)
put her in charge
say charge
her into is 3
put her      into varA
put her into into varB
say varA
say varB
say ""

say "array"
Alice is 2
rock the array with her, her times, and her into
say the array
say the array at 0
say the array at 1
say the array at 2
say ""

say "increment and decrement"
two is 2
build her up
say it
build her up up
say it
knock her down
say it
knock her down, down
say it
say ""

say "function return"
functionA takes zip
    her back is sore
    her arms are tired
    give her back (her = her arms = 5)

say functionA taking nothing

functionB takes her back
    her back is sore
    her arms are tired
    give her back back (her back = 4)

say functionB taking nothing

(
rock her times with 12
say her times at 0
say her times at 1
)

(

rock her times with 10
roll her times into her times
split her times into her times
split her times into her times with her times
join her times with ","
join her times into her times with ","
cast her times into her times with her times
)




(
a rock is small
the rock is huge
her heart is big

say her heart times 4
say her times 5

her times are great
say her times
say her times 2
say her times times 3

say the times times 6

say her lies times

if her times 4 is 12
say yes
else
say no

if her heart is as big as the rock
say yes
else
say no
)
