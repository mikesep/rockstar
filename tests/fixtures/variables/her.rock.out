pronoun
21
0
herherher
9
9
15
33

common variable
5
false
false
false
false
15
25

assignment
156
3
3

array
3
2
5
3

increment and decrement
3
5
4
2

function return
5
4
